CMakeLists.txt

```diff
@@ -15,24 +15,25 @@ function(append_path_if_set VAR_NAME INCLUDE_LIST LIBRARY_LIST)
     # Variable set locally or via the command line
     if(${VAR_NAME})
         message(STATUS ${VAR_NAME} " set via command line to: " ${${VAR_NAME}})
-        set(${INCLUDE_LIST} ${${INCLUDE_LIST}} "${${VAR_NAME}}/include" PARENT_SCOPE)
-        set(${LIBRARY_LIST} ${${LIBRARY_LIST}} "${${VAR_NAME}}/lib" PARENT_SCOPE)
+        list(APPEND ${INCLUDE_LIST} "${${VAR_NAME}}/include" PARENT_SCOPE)
```

Comment:
This change does not work... PARENT_SCOPE is not recognized for list CMake operation.
It could work if the function becomes a macro because variables in cmake macros have a global scope:

`macro(append_path_if_set VAR_NAME INCLUDE_LIST LIBRARY_LIST)

    # Variable set locally or via the command line
    if(${VAR_NAME})
        message(STATUS ${VAR_NAME} " set via command line to: " ${${VAR_NAME}})
        list(APPEND ${INCLUDE_LIST} "${${VAR_NAME}}/include")
        list(APPEND ${LIBRARY_LIST} "${${VAR_NAME}}/lib")
    else()
        set(${VAR_NAME} $ENV{${VAR_NAME}} )

        # Variable may be set in the environment
        if(${VAR_NAME})
            message(STATUS ${VAR_NAME} " set via env to: " ${${VAR_NAME}})
            list(APPEND ${INCLUDE_LIST} "$ENV{${VAR_NAME}}/include")
            list(APPEND ${LIBRARY_LIST} "$ENV{${VAR_NAME}}/lib")            
        endif()
    endif()
endmacro()`

There might be some better ways of doing it... I don't consider myself as a CMake expert... ;-)